yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -485,7 +485,7 @@ private[spark] class ApplicationMaster(
             e.getCause match {
               case _: InterruptedException =>
                 // Reporter thread can interrupt to stop user class
-              case e: Exception =>
+              case e: Throwable =>
```

Comment:
`e` still shadows the `e: InvocationTargetException` above. I know it was already that way in the code and it works as intended this way. However while we're here I wonder if it's worth calling this variable `cause`? then it's obvious that `cause` is being rethrown.